lib/iex/lib/iex/autocomplete.ex

```diff
@@ -440,15 +442,23 @@ defmodule IEx.Autocomplete do
   end
 
   defp get_module_types(mod) do
-    cond do
-      not ensure_loaded?(mod) ->
+    with true <- ensure_loaded?(mod),
+         docs when not is_nil(docs) <- get_docs(mod, [:type]) do
+      Enum.map(docs, &extract_name_and_arity/1)
+    else
+      false ->
         []
 
-      docs = get_docs(mod, [:type]) ->
-        Enum.map(docs, &extract_name_and_arity/1)
+      nil ->
+        case Typespec.fetch_types(mod) do
+          {:ok, specs} ->
+            Enum.map(specs, fn {_kind, {name, _, _}} ->
+              {name, 0}
```

Comment:
we shouldn't hardcode arity to zero, instead you can do:

```elixir
Enum.map(specs, fn {_kind, {name, _, args}} ->
  {name, length(args)}
```

Reply:
Is this documented somewhere? I tried to guess what is what in the AST because I found no resources about the structure.

Reply:
I don't think it is, maybe somewhere in Erlang docs. If Code.Typespec module ever becomes public, I think that'd be a good place to document it.